frontend/tests/controllers/QualityNominationTest.php

```diff
@@ -2451,4 +2451,144 @@ public function testCanonicalizeTags() {
         );
     }
+
+    /**
+     * A PHPUnit data provider for all the tests that can accept a status.
+     *
+     * @return list<array{0: string, 1: int, 2: int, 3: string, 4:array<int:boolean>, 5: boolean }>
+     */
+    public function qualityNominationsDemotionStatusApiUpdateCaseProvider(): array {
+        return [
+            ['warning', \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING,
+            'qualityNominationProblemHasWarning',
+            [\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_BANNED => false,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED => false,
+            \OmegaUp\ProblemParams::VISIBILITY_PRIVATE => false,\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_WARNING => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING => true,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC => false,
+            \OmegaUp\ProblemParams::VISIBILITY_PROMOTED => false], true
+            ],
+            ['warning', \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING,
+            'qualityNominationProblemHasWarning',
+            [\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_BANNED => true,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PRIVATE => true,\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_WARNING => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING => true,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PROMOTED => true], false
+            ],
+            ['banned', \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED,
+            'qualityNominationProblemHasBeenBanned',
+            [\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_BANNED => false,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PRIVATE => false,\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_WARNING => false,
+            \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING => false,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC => false,
+            \OmegaUp\ProblemParams::VISIBILITY_PROMOTED => false], true,
+            ],
+            ['banned', \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED,
+            'qualityNominationProblemHasBeenBanned',
+            [\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_BANNED => true,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PRIVATE => true,\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_WARNING => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING => true,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PROMOTED => true], false,
+            ]
+        ];
+    }
+
+    /**
+     * Check that a non-reviewer user cannot change the visibility that is not PRIVATE_WARNING or PUBLIC_WARNING
+     * of his problems with a demotion qualitynomination but only reviewer can change any visibility.
```

Comment:
nit: hay que intentar que el código quepa en el diff de GitHub (~100 caracteres por línea):

```suggestion
     * Check that a non-reviewer user cannot change the visibility that is not PRIVATE_WARNING
     * or PUBLIC_WARNING of his problems with a demotion qualitynomination but only reviewer
     * can change any visibility.
```

Reply:
listo